src/postProposalVersions.ts

```diff
@@ -0,0 +1,320 @@
+}
+
+const args = argParse<Args>({
+  inputFile: String,
+  applicantColumnName: String,
+  applicationFormId: Number,
+  proposalExternalIdColumnName: String,
+  bearerToken: String,
+  apiUrl: String,
+});
+
+const csvInput = createReadStream(args.inputFile, 'utf8');
+const {
+  applicantColumnName,
+  applicationFormId,
+  proposalExternalIdColumnName,
+  bearerToken,
+  apiUrl,
+} = args;
+
+const headers = {
+  accept: 'application/json',
+  authorization: `Bearer ${bearerToken}`,
+};
+
+const requestTimeoutMs = 60000;
+
+const jsonSubstring = (json: object) => JSON.stringify(json).substring(0, 40);
+
+const getOrPostApplicantByExternalId = async (applicantExternalId: string) => {
+  let applicant: Applicant | undefined;
+  let finalApplicant: Applicant;
+
+  const applicants = (
+    await axios.get<Applicant[]>(
+      `${apiUrl}/applicants`,
+      {
+        timeout: requestTimeoutMs,
+        headers,
+      },
+    )
+  ).data;
+
+  if (applicants.length > 0) {
+    [applicant] = applicants.filter((a) => a.externalId === applicantExternalId);
+  }
+
+  if (applicant === undefined) {
+    try {
+      applicant = (
+        await axios.post<Applicant>(
+          `${apiUrl}/applicants`,
+          { externalId: applicantExternalId },
+          {
+            timeout: requestTimeoutMs,
+            headers,
+          },
+        )
+      ).data;
+    } catch (error: unknown) {
+      if (error instanceof AxiosError
+        && error.response !== undefined
+        && error.response.status === 409) {
+        // Get the applicants again.
+        const applicantsAgain = (
+          await axios.get<Applicant[]>(
+            `${apiUrl}/applicants`,
+            {
+              timeout: requestTimeoutMs,
+              headers,
+            },
+          )
+        ).data;
+
+        if (applicantsAgain.length > 0) {
+          [applicant] = applicantsAgain.filter((a) => a.externalId === applicantExternalId);
+        }
+      }
+    }
+  }
+
+  if (applicant === undefined) {
+    throw new Error('Could not GET or POST an applicant.');
+  } else {
+    finalApplicant = applicant;
+  }
+
+  return finalApplicant;
+};
+
+const getOrPostProposal = async (
+  proposals: Proposal[],
+  opportunityId: number,
+  applicant: Applicant,
+  proposalExternalId: string,
+) => {
+  let proposal: Proposal | undefined;
+
+  // Pick the only existing proposal with given opportunity id, applicant id, and external id.
+  // This should work because there is a unique constraint across these three fields.
+  if (proposals.length > 0) {
+    [proposal] = proposals.filter(
+      (p) => (p.opportunityId === opportunityId
+        && p.applicantId === applicant.id
+        && p.externalId === proposalExternalId
+      ),
+    );
+  }
+
+  // If the proposal did not exist, create it.
+  if (proposal === undefined) {
+    console.log(`No existing proposal for opportunityId=${opportunityId}, applicantId=${applicant.id}, and externalId=${proposalExternalId}`);
```

Comment:
Obviously these console statements can be replaced after a logger is present. These are not the issue.

Reply:
After @jasonaowen merges a minimal logger setup, I will update this PR with logger statements instead of `console.log`.

Reply:
See PR #48.

Reply:
Added the logger. I started with my child logger setup, but then that turned out to be redundant when I'm explicitly running this one script. I suppose if it called any functions from other files it would be useful again. For now I'm skipping that step.